includes/Core/Conversion_Tracking/Conversion_Tracking.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Modules\Conversion_Tracking
+ *
+ * @package   Google\Site_Kit\Core\Modules
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Conversion_Tracking;
+
+use Google\Site_Kit\Context;
+
+/**
+ * Class for managing conversion tracking.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Conversion_Tracking {
+
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * Conversion_Event_Provider_Registry instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Conversion_Event_Provider_Registry
+	 */
+	private $registry;
+
+	/**
+	 * Supported conversion event providers.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	public static $conversion_event_providers = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context $context Plugin context.
+	 */
+	public function __construct( Context $context ) {
+		$this->context = $context;
+	}
+
+	/**
+	 * Registers the class functionality.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		$active_conversion_event_providers = $this->get_active_conversion_event_providers();
+
+		array_walk(
+			$active_conversion_event_providers,
+			function( Conversion_Events_Provider $active_conversion_event_provider ) {
+				$script_asset = $active_conversion_event_provider->register_script();
+				$script_asset->enqueue();
+			}
+		);
+	}
+
+	/**
+	 * Gets the instances of active conversion event providers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of active Conversion_Events_Provider instances.
+	 */
+	public function get_active_conversion_event_providers() {
+		$active_conversion_event_providers = array();
+		$conversion_event_provider_classes = $this->get_registry()->get_all();
+
+		foreach ( $conversion_event_provider_classes as $conversion_event_provider_class ) {
+			$instance = new $conversion_event_provider_class( $this->context );
+
+			if ( $instance->is_active() ) {
+				$active_conversion_event_providers[ $instance::CONVERSION_EVENT_PROVIDER_SLUG ] = $instance;
```

Comment:
FWIW, I see you did correctly defined this slug in your test, but maybe just forgot to include it in the abstract class, I feel we need to make it required, as it is required.

Reply:
Since it is a constant holding dynamic string for slugs, `Conversion_Event_Provider` extending it won't be able to redeclare it, since constant can't be overwritten. So individual implementation will handle defining that constant.

Reply:
@zutigrm fair enough, but then we probably should check for the existence of this constant the same time we check for `$instance->is_active()` or will the instance never be active without the constant? I am just conscious of the fact that we're keying by a value we assume to be present.